src/analysis/retail/priest/shadow/modules/features/getUptimeGraph.tsx

```diff
@@ -0,0 +1,158 @@
+import BaseChart, { formatTime } from 'parser/ui/BaseChart';
+import { AutoSizer } from 'react-virtualized';
+import { VisualizationSpec } from 'react-vega';
+
+export interface UptimeHistoryEntry {
+  timestamp: number;
+  uptimePct: number;
+}
+
+interface GraphData {
+  timestamp: number;
+  activeTimePercentage: number;
+}
+
+const GRAPH_SAMPLE_COUNT = 2; // Show value for every Nth GCD in the graph.
+const GRAPH_MAX_Y = 1; // The maximum value on the Y axis. Used to prevent the graph skyrocketing.
+
+export default function getUptimeGraph(
```

Comment:
my only comment is that if this is a straight copy of the ele graph, we should prob share it. other than that, this is good

Reply:
I think I've moved getUptimeGraph to a shared location for both Ele and Shadow priest.